crates/breez-sdk/core/src/sdk.rs

```diff
@@ -505,6 +649,26 @@ impl BreezSdk {
         );
         Ok(transfer)
     }
+
+    fn ensure_default_spark_private_mode_applied_if_on(&self) -> Result<(), SdkError> {
+        if !*self.default_user_settings_applied_watcher.borrow()
```

Comment:
I see an edge case where if the real-time sync is not available for some reason, maybe connectivity/server issues, an SDK without initial settings applied would be blocked. Are we ok with that?

Reply:
Right @dangeross  but it will only do so if the current config attempting to set the mode to private. It seems something we can't avoid based on the product decision?